parquet-format-structures/src/main/java/org/apache/parquet/format/BlockCipher.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.format;
+
+import java.io.IOException;
+
+public interface BlockCipher{
+
+  public interface Encryptor{
+    /**
+     * Encrypts the plaintext.
+     * Make sure the returned contents starts at offset 0 and fills up the byte array.
```

Comment:
(nit) This "make sure" sound like the caller has to make sure to fulfill these conditions, when in fact it is the method implementation that does this.

```suggestion
     * The ciphertext starts at offset 0 of the return value and fills up the entire byte array.
```

Similarly in the other methods below.

Reply:
SGTM.